src/main/java/com/google/sps/servlets/SearchServlet.java

```diff
@@ -55,46 +55,121 @@
    */
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response)
-    throws ServletException, IOException {
-    String stringToMatch = request
-      .getParameter("searchterm")
-      .replaceAll("[^a-zA-Z0-9]+", " ")
-      .trim();
-    request.setAttribute("dataSearched", stringToMatch);
+    throws ServletException, IOException, NullPointerException {
+    List<Recipe> recipesListToReturn = new ArrayList<>();
+    Integer timeValue;
+    String stringToMatch = getParameter("searchterm", request);
+    String difficulty = getParameter("difficulty", request);
+    String time = getParameter("time", request);
+    HashMap<String, String> difficultiesMap = createMapOfDifficulties();
+    HashMap<Integer, String> timeIntervalsMap = createMapOfTimeIntervals();
 
-    List<Recipe> recipesListToReturn = recipesMatching(
-      request,
-      response,
-      stringToMatch
+    request.setAttribute("dataSearched", stringToMatch);
+    request.setAttribute("difficultySearchedKey", difficulty);
+    request.setAttribute("timeSearchedKey", time);
+    request.setAttribute(
+      "difficultySearchedValue",
+      difficultiesMap.get(difficulty)
     );
+    request.setAttribute("difficultyList", difficultiesMap);
+    request.setAttribute("cookingTimeList", timeIntervalsMap);
 
+    stringToMatch = trimAndRemoveCommas(stringToMatch);
+    difficulty = trimAndRemoveCommas(difficulty);
+    try {
+      timeValue = Integer.valueOf(trimAndRemoveCommas(time));
```

Comment:
Why are you trimming here if these are coming from selects?

Reply:
Because the user can insert trash values inside the request directly from the link.